kotlinx-coroutines-core/common/test/channels/ClosedChannelTest.kt

```diff
@@ -0,0 +1,196 @@
+package kotlinx.coroutines.channels
+
+import kotlinx.coroutines.testing.*
+import kotlin.test.*
+
+class ClosedChannelTest : TestBase() {
+    /**
+     * Iterator.
+     */
+
+    @Test
+    fun testIteratorClosedHasNextFalse() = runTest {
+        TestChannelKind.entries.forEach { kind ->
```

Comment:
Proposal:
```kotlin
    fun TestChannelKind.Companion.runTestForEach(
        testBody: suspend CoroutineScope.(TestChannelKind) -> Unit
    ): TestResult = runTest {
        TestChannelKind.entries.forEach { kind ->
            testBody(kind)
        }
    }
```
May be more convenient, with how often this pattern occurs.

Reply:
It looks ugly, I'd rather not. See the last commit. Шило на мыло. Revert?

Reply:
The approach with `ChannelTestBase` does indeed look too complicated and indirect for such small gains, so yes, I agree it's not a win. That wasn't what I had in mind, though. My thought was to introduce only the `fun TestChannelKind.Companion.runTestForEach` extension in `kotlinx.coroutines.channels` in the test source set (or `fun TestBase.runTestForEachKind` if you prefer).

In any case, this is just a suggestion, please ignore it if you don't find it valuable.

Reply:
It's not that now I need to inherit from `ChannelTestBase`. It's actually the way it looks in code, and it will look identical with your approach as well. Reverted.